tests/conftest.py

```diff
@@ -88,39 +83,29 @@ async def async_generator():
         mock.return_value = async_generator()
 
     mock.set_generator_values = set_generator_values
-    return mock
-
-
-@pytest.fixture
-def mock_collect_user_input(mocker):
-    mock_method = mocker.MagicMock()
-    mocker.patch.object(UserInputManager, "collect_user_input", new=mock_method)
 
-    return mock_method
+    return mock
 
 
-@pytest.fixture
+@pytest.fixture(scope="function")
 def mock_setup_api_key(mocker):
-    mocker.patch("mentat.app.setup_api_key")
-    mocker.patch("mentat.conversation.is_model_available")
+    mocker.patch("mentat.llm_api.setup_api_key")
```

Comment:
This is wrong; mocking works by mocking the IMPORT of the function, not the actual function itself (which is extremely counterintuitive, but that's how it is), so you need to mock it in the file that it's imported/used in; in this case, mentat.session.setup_api_key, and mentat.conversation.is_model_available